Pokedex Cog

Redjumpman opened the pull request:
The code is crazy I know. I've only been learning python for the past month, but I'm open to suggestions. It DOES however work. So please have fun with it, and feel free to modify it. Current list of features:
- !pokedex pokemon '_pokemon name_'  - retrieves a pokemon's pokedex data
- !pokedex moveset '_generation#_' '_pokemon name_' - retrieves a pokemon's moveset by generation
- !pokedex stats '_pokemon name_' - retrieves a pokemon's base stats along with min and max at Lv 100
- !pokedex item "_item name_' - Retrieves an item's effect
- !pokedex location '_pokemon name_'  - Retrieves a pokemon's catch location by game version
- !pokedex evolution '_pokemon name_' - Will display evolution chain and requirements to evolve

Planned to do list:
- [x] ~~Search for pokemon's catch location(s)~~
- [x] ~~Evolution chain search~~
- [ ] TM/HM Search list (can do this with item search currently)
- [ ] Add flavor text to pokedex search
- [ ] Condense code
- [x] ~~Remove numpy dependency~~

**Edit**. As a note it does require 3 libs: BS4, ~~Numpy,~~ and Tabulate. I included the installation instructions for those libraries in my JSON.

Redjumpman committed:
Added files via upload

Redjumpman committed:
Added files via upload

Redjumpman committed:
Update pokedex.py

Redjumpman commented:
Cleaned up the code and edited a few things. Should look much neater. Still needs some love, but it works.

irdumbs commented:
<3 we will take a look and give you feedback.
please bug us/me in the Red server.
I approve of this cog

Redjumpman committed:
Update info.json

Redjumpman committed:
Update pokedex.py

Redjumpman committed:
Update pokedex.py

Redjumpman committed:
Update pokedex.py

Added catch locations as a searchable feature.

Redjumpman committed:
Added evolution chain search.

- added evolution chain searches
- added better error checking
- added roman numerals for move sets. can now use both numbers and roman numerals :D

Redjumpman committed:
Update pokedex.py

Fixed errors when users typed in an incorrect pokemon or generation

Redjumpman committed:
Delete .gitignore

Redjumpman committed:
Delete README.md
